pkg/registry/core/service/strategy.go

```diff
@@ -186,11 +186,10 @@ func (strategy svcStrategy) serviceDualStackFieldsInUse(svc *api.Service) bool {
 		return false
 	}
 
-	ipFamilyPolicyInUse := svc.Spec.IPFamilyPolicy != nil && *(svc.Spec.IPFamilyPolicy) != api.IPFamilyPolicySingleStack
+	ipFamilyPolicyInUse := svc.Spec.IPFamilyPolicy != nil
 	// we have to check if user is using a family not configured on cluster in single stack gate off mode
 	// in order to return correct errors.
-	ipFamiliesInUse := len(svc.Spec.IPFamilies) == 2 ||
-		(len(svc.Spec.IPFamilies) == 1 && svc.Spec.IPFamilies[0] != strategy.ipFamilies[0])
+	ipFamiliesInUse := len(svc.Spec.IPFamilies) > 1 || (len(svc.Spec.IPFamilies) > 0 && svc.Spec.IPFamilies[0] != strategy.ipFamilies[0])
```

Comment:
why `> 1` and not `> 0` ?  The only places I can see that set ipFamiles are in tryDefaultValidateServiceClusterIPFields() which early-returns if the gate is closed.  So we should never see any values here unless the feature is enabled.